src/Components/Admins/Form/index.jsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, useEffect } from 'react';
+import styles from './form.module.css';
+
+const Form = (props) => {
+  const [formData, setFormData] = useState({
+    firstName: '',
+    lastName: '',
+    dni: '',
+    phone: '',
+    email: '',
+    city: '',
+    password: ''
+  });
+
+  const [isEditing, setIsEditing] = useState(false);
+
+  useEffect(() => {
+    if (props.selectedAdmin) {
+      setIsEditing(true);
+      setFormData({
+        firstName: props.selectedAdmin.firstName,
+        lastName: props.selectedAdmin.lastName,
+        dni: props.selectedAdmin.dni,
+        phone: props.selectedAdmin.phone,
+        email: props.selectedAdmin.email,
+        city: props.selectedAdmin.city,
+        password: props.selectedAdmin.password
+      });
+    } else {
+      setIsEditing(false);
+      setFormData({
+        firstName: '',
+        lastName: '',
+        dni: '',
+        phone: '',
+        email: '',
+        city: '',
+        password: ''
+      });
+    }
+  }, [props.selectedAdmin]);
+
+  const onChange = (e) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  const onSubmit = (e) => {
+    e.preventDefault();
+    if (isEditing) {
+      props.editAdmin(formData, props.selectedAdmin._id);
+    } else {
+      props.addAdmin(formData);
+    }
+    setFormData({
```

Comment:
I think that this reset of the fields is unnecessary, once the request is successfully done the form should hide. If you enter an invalid value while adding an admin, the fields values reset.